metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java

```diff
@@ -167,20 +172,10 @@ synchronized void start(long initialDelay, long period, TimeUnit unit, Runnable
             throw new IllegalArgumentException("Reporter already started");
         }
 
-        this.scheduledFuture = getScheduledFuture(initialDelay, period, unit, runnable);
+        this.scheduledFuture = this.getScheduledFuture.schedule(initialDelay, period, unit, runnable, this.executor);
     }
 
 
-    /**
-     * Schedule the task, and return a future.
-     *
-     * @deprecated Use {@link #getScheduledFuture(long, long, TimeUnit, Runnable, ScheduledExecutorService)} instead.
-     */
-    @SuppressWarnings("DeprecatedIsStillUsed")
-    @Deprecated
-    protected ScheduledFuture<?> getScheduledFuture(long initialDelay, long period, TimeUnit unit, Runnable runnable) {
-        return getScheduledFuture(initialDelay, period, unit, runnable, this.executor);
-    }
```

Comment:
I am aware this is a breaking change. Unfortunately, I did not find a way to make this function still relevant while passing the functional interface
Additionally, I think it is unlikely someone is overriding this method because it can make sense only with an executor.

What do you think?
Something obvious I missed in keeping this method for retro compatibility, while still making it relevant?